plugins/woocommerce/client/legacy/js/admin/meta-boxes-order.js

```diff
@@ -1585,11 +1588,112 @@ jQuery( function ( $ ) {
+				metakey_select = $(this);
+				select2_args = {
+					allowClear: !!$(this).data('allow_clear'),
+					placeholder: $(this).data('placeholder'),
+					escapeMarkup: function (m) {
+						return m;
+					},
+					ajax: {
+						url: wc_enhanced_select_params.ajax_url,
+						dataType: 'json',
+						delay: 500,
+						data: function (params) {
+							return {
+								order_id: $(this).data('order_id'),
+								action: 'woocommerce_json_search_order_metakeys',
+								security: wc_enhanced_select_params.search_order_metakeys_nonce
+							};
+						},
+						success: function (data) {
+							var terms = [];
+							if (data) {
+								$.each(data, function (id, term) {
+									terms.push({
+										id: term,
+										text: term,
+									});
+								});
+							}
+							// Reinitialize with the loaded data to avoid continued ajax requests when searching since
+							// we are not using the search term to filter the list on the backend.
+							select2_args.data = terms;
+							delete select2_args.ajax;
+							metakey_select.selectWoo(select2_args).select2('open');
+						},
+						cache: true
+					},
+					language: {
+						errorLoading: function () {
+							// Workaround for https://github.com/select2/select2/issues/4355 instead of i18n_ajax_error.
+							return wc_enhanced_select_params.i18n_searching;
+						},
+						inputTooLong: function (args) {
+							var overChars = args.input.length - args.maximum;
+
+							if (1 === overChars) {
+								return wc_enhanced_select_params.i18n_input_too_long_1;
+							}
+
+							return wc_enhanced_select_params.i18n_input_too_long_n.replace('%qty%', overChars);
+						},
+						inputTooShort: function (args) {
+							var remainingChars = args.minimum - args.input.length;
+
+							if (1 === remainingChars) {
+								return wc_enhanced_select_params.i18n_input_too_short_1;
+							}
+
+							return wc_enhanced_select_params.i18n_input_too_short_n.replace('%qty%', remainingChars);
+						},
+						loadingMore: function () {
+							return wc_enhanced_select_params.i18n_load_more;
+						},
+						maximumSelected: function (args) {
+							if (args.maximum === 1) {
+								return wc_enhanced_select_params.i18n_selection_too_long_1;
+							}
+
+							return wc_enhanced_select_params.i18n_selection_too_long_n.replace('%qty%', args.maximum);
+						},
+						noResults: function () {
+							return wc_enhanced_select_params.i18n_no_matches;
+						},
+						searching: function () {
+							return wc_enhanced_select_params.i18n_searching;
+						}
+					}
+
+				};
+
+				// Work around to deal with the lack of responsive support from Select2 until a better replacement can be integrated.
+				let resizeTimer;
+				$( window ).on( 'resize', function () {
+					clearTimeout( resizeTimer );
+					resizeTimer = setTimeout( function () {
+						metakey_select.selectWoo( select2_args );
+					}, 300 );
```

Comment:
requestionAnimationFrame would likely be better here and more performant unless the delay is intended from a UX perspective